diff --git a/openapi/src/apis/mod.rs b/openapi/src/apis/mod.rs
index 18c0d9d..9347831 100644
--- a/openapi/src/apis/mod.rs
+++ b/openapi/src/apis/mod.rs
@@ -103,7 +103,11 @@ pub fn parse_deep_object(prefix: &str, value: &serde_json::Value) -> Vec<(String
 
 pub mod authorization_api;
 pub mod circuits_api;
-pub mod internal_api;
+#[allow(dead_code)]
+mod internal_api;
+pub use internal_api::{
+    circuit_download, circuit_status, proof_status, user_me, CircuitStatusError, ProofStatusError,
+};
 pub mod proofs_api;
 pub mod token_api;
 
